pkg/agent/agent.go

```diff
@@ -243,14 +250,36 @@ func (a *Agent) bootstrap() error {
 		a.baseSVIDKey = key
 
 		// If we're here, we need to attest/Re-attest
-		regEntryMap, err := a.attest()
+		regEntries, err := a.attest()
 		if err != nil {
 			return err
 		}
-		err = a.FetchSVID(regEntryMap, a.BaseSVID, a.baseSVIDKey)
-		if err != nil {
-			return err
+		serverId := url.URL{
+			Scheme: "spiffe",
+			Host:   a.config.TrustDomain.Host,
+			Path:   path.Join("spiffe", "cp"),
 		}
+		cmgrConfig := &cache.MgrConfig{
+			ServerCerts:    a.serverCerts,
+			ServerSPIFFEID: serverId.String(),
+			ServerAddr:     a.config.ServerAddress.String(),
+
+			BaseSVID:       a.BaseSVID,
+			BaseSVIDKey:    a.baseSVIDKey,
+			BaseRegEntries: regEntries,
+			Logger:         a.config.Log,
+		}
+
+		a.CacheMgr, err = cache.NewManager(a.ctx, cmgrConfig)
+
+		a.CacheMgr.Init()
+		go func() {
+			<-a.CacheMgr.Done()
```

Comment:
Will this unblock on an error condition too or only a clean shutdown? Should we shutdown the agent?

Reply:
Yes  `CacheMgr.Done()` will return when `doneCh` is closed which happens when the Init() method returns